src/Unirest/Request.php

```diff
@@ -558,4 +558,18 @@ private static function getHeaderString($key, $val)
         $key = trim(strtolower($key));
         return $key . ': ' . $val;
     }
+
+    /**
+     * @param array $existing_options
+     * @param array $new_options
+     * @return array
+     */
+    private static function mergeCurlOptions(&$existing_options, $new_options)
```

Comment:
you can just use the `+` operator here instead of walking through the array ...

interesting to learn that `array_merge` seems to only merge arrays that have the same **string** keys